### PR TITLE
Feature/simple logging

### LINKED ISSUE #76: Give status information during startup of TestRunner
Currently the startup time of cli is quite long (several seconds), because there are many things done before the TestRunner is started.
We should give some status information about what's done.
We should also include a new command-line parameter to suppress these status information (--silent)

### LINKED ISSUE #89: Possibility to add debug-logging with several helpful information
To make support/debugging easier (see https://github.com/utPLSQL/utPLSQL-cli/issues/88) we should have an option to enable debug-logging. 
Information that should be included:
- Java-Version
- utPLSQL/cli/java-api version
- ORACLE_HOME value

### PR DESCRIPTION
**Pull request by pesse**
Adds some simple logging to cli. Does basically this:

```
######################### utPLCSL cli #########################
#                                                             #
#   utPLSL-cli 3.1.3-SNAPSHOT.local                           #
#   utPLSQL-java-api 3.1.3-SNAPSHOT.123                       #
#   Java-Version: 1.8.0_171                                   #
#   ORACLE_HOME: C:\app\client\sani\product\12.2.0\client_1   #
#                                                             #
#   Thanks for testing!                                       #
#                                                             #
###############################################################

Successfully connected to database. UtPLSQL core: v3.1.2.2130
Oracle-Version: 12.2.0.1.0
Running tests now.
--------------------------------------
```

Basic logging can be turned off via `-q` or `--quiet`
Additional, extensive, non-pretty logging can be turned on via `-d` or `--debug`

Fixes #89 
Fixes #76

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,36 +20,42 @@
...
+          <version>2.7.2</version>
+          <scope>compile</scope>
+      </dependency>
+      <dependency>
+          <groupId>javax.xml.bind</groupId>
+          <artifactId>jaxb-api</artifactId>
+          <version>2.3.0</version>
+      </dependency>
+      <dependency>
+          <groupId>ch.qos.logback</groupId>
```
Comment (Pazus) [inline pom.xml]: What's the reason behind choosing logback instead onf sl4j?

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.ConsoleAppender;
+import com.zaxxer.hikari.HikariDataSource;
+import org.slf4j.LoggerFactory;
+import org.utplsql.api.TestRunner;
+
+class LoggerConfiguration {
+
```
Comment (Pazus) [inline src/main/java/org/utplsql/cli/LoggerConfiguration.java]: if class is non-instantiable pleaso provide private constructor with UnsupportedOperationException
```suggestion
    private LoggerConfiguration() {
       throw new UnsupportedOperationException();
    }
```

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    static void configure(boolean silent, boolean debug) {
```
Comment (Pazus) [inline src/main/java/org/utplsql/cli/LoggerConfiguration.java]: Why do we use two parameters here? Are all 4 combination of their values valid? Maybe we should accept 3-valued enum? 
```
SILENT, DEFAULT, DEBUG
```

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        else
+            configureDefault();
+    }
+
+    private static void configureSilent() {
+        Logger root = (Logger)LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        root.setLevel(Level.OFF);
+    }
+
+    private static void configureDefault() {
```
Comment (Pazus) [inline src/main/java/org/utplsql/cli/LoggerConfiguration.java]: This kind of manual logging configuration seems a bit strange for me...
Isn't default simply equals to INFO logging level, Debug for debug and silent for non?

File: src/main/java/org/utplsql/cli/RunCommand.java [hunk 5]
```diff
@@ -221,6 +228,48 @@ public String getCommand() {
...
+        try (Connection conn = dataSource.getConnection()) {
+
+            // Check if orai18n exists if database version is 11g
+            RunCommandChecker.checkOracleI18nExists(conn);
+
+            // First of all do a compatibility check and fail-fast
+            compatibilityProxy = checkFrameworkCompatibility(conn);
+
+            logger.info("Successfully connected to database. UtPLSQL core: " + compatibilityProxy.getDatabaseVersion());
+            logger.info("Oracle-Version: {}", new DefaultDatabaseInformation().getOracleVersion(conn));
```
Comment (Pazus) [inline src/main/java/org/utplsql/cli/RunCommand.java]: is it possible to provide it as lambda so we don't execute it if not needed?

File: src/test/java/org/utplsql/cli/TestRunCommandChecker.java [hunk 6]
```diff
@@ -15,7 +15,7 @@ public void getCheckFailOnErrorMessage()
         assertNotNull(RunCommandChecker.getCheckFailOnErrorMessage(2, new Version("3.0.0")));
         assertNotNull(RunCommandChecker.getCheckFailOnErrorMessage(2, new Version("3.0.1")));
         assertNotNull(RunCommandChecker.getCheckFailOnErrorMessage(2, new Version("3.0.2")));
-        assertNull(RunCommandChecker.getCheckFailOnErrorMessage(2, new Version("3.0.3")));
+        assertNull(RunCommandChecker.getCheckFailOnErrorMessage(2, new Version("3.0.3.1266")));
```
Comment (Pazus) [inline src/test/java/org/utplsql/cli/TestRunCommandChecker.java]: What's the reason for the change?

**Review by Pazus — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (pesse) [reply pom.xml]: slf4j is the logging framework (which is used), but it needs an implementation to run (which is logback because most lightweight with still the features needed)

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java (hunk 3 above)
Comment (pesse) [reply src/main/java/org/utplsql/cli/LoggerConfiguration.java]: Reason is to make it easy to just pass the parameters of the RunCommand to the LoggerConfigurator. 
Enum would increase clarity of the LoggerConfigurator class taken alone but not its usage. This way it's clear that logging is depenent on two flags.
If we can increase clarity on the fact that silent is superior to debug, I'd appreciate it.

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java (hunk 4 above)
Comment (pesse) [reply src/main/java/org/utplsql/cli/LoggerConfiguration.java]: No, because I don't want the default pattern-layout mess to appear on normal informational output. And yes, it's inconvenient that I have to touch so many things, but that's the only way I found to achieve what I want.

File: src/main/java/org/utplsql/cli/RunCommand.java (hunk 5 above)
Comment (pesse) [reply src/main/java/org/utplsql/cli/RunCommand.java]: Please clarify, I don't get it

File: src/test/java/org/utplsql/cli/TestRunCommandChecker.java (hunk 6 above)
Comment (pesse) [reply src/test/java/org/utplsql/cli/TestRunCommandChecker.java]: Change in the API (where I put the exact numbers for the feature, which makes sense)

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java (hunk 3 above)
Comment (Pazus) [reply src/main/java/org/utplsql/cli/LoggerConfiguration.java]: I think we should. If non flags are passed then default logging is enabled, if silent - non, if debug - debug. And then we end up with 3 levels, set on application level and non of other classes should think about it. Classes just log using appropriate log level (`logger.info`, `logger.debug`).

File: src/main/java/org/utplsql/cli/RunCommand.java (hunk 5 above)
Comment (Pazus) [reply src/main/java/org/utplsql/cli/RunCommand.java]: something like this: 
```
logger.debug("I am logging that {} happened.", () -> compute());
```
or maybe 
```
logger.debug(() -> "I am logging that "+compute()+" happened.")
```
so db call gather information is only done when logging level is low enough.

File: src/main/java/org/utplsql/cli/LoggerConfiguration.java (hunk 3 above)
Comment (pesse) [reply src/main/java/org/utplsql/cli/LoggerConfiguration.java]: I'm not sure if I get you completely right.
You'd like to introduce a 3-leveled ENUM and shift the logic which configuration to take to the RunCommand?
The classes themselves already use appropriate log level, the purpose of the configuration is to either hide or show the log information.

File: src/main/java/org/utplsql/cli/RunCommand.java (hunk 5 above)
Comment (pesse) [reply src/main/java/org/utplsql/cli/RunCommand.java]: Neat, didn't know this usage.

Comment (pesse) [reply src/main/java/org/utplsql/cli/RunCommand.java]: It's not possible at the moment.
See https://jira.qos.ch/browse/SLF4J-371

**Review by Pazus — APPROVED** [review]
